src/com/goide/codeInsight/intention/GoMoveToStructAssignmentAction.java

```diff
@@ -0,0 +1,198 @@
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.goide.codeInsight.intention;
+
+import com.goide.psi.*;
+import com.goide.psi.impl.GoElementFactory;
+import com.goide.psi.impl.GoPsiImplUtil;
+import com.intellij.codeInsight.intention.PsiElementBaseIntentionAction;
+import com.intellij.openapi.command.WriteCommandAction;
+import com.intellij.openapi.editor.Editor;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.util.Condition;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiWhiteSpace;
+import com.intellij.psi.util.PsiTreeUtil;
+import com.intellij.util.IncorrectOperationException;
+import com.intellij.util.containers.ContainerUtil;
+import org.jetbrains.annotations.Contract;
+import org.jetbrains.annotations.Nls;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.List;
+
+public class GoMoveToStructAssignmentAction extends PsiElementBaseIntentionAction {
+  public static final String NAME = "Move field initialization to struct assignment";
+
+  @Override
+  public boolean isAvailable(@NotNull Project project, Editor editor, @NotNull PsiElement element) {
+    if (!element.isValid() || !element.isWritable()) return false;
+    GoAssignmentStatement assignment = PsiTreeUtil.getParentOfType(element, GoAssignmentStatement.class, false);
+    if (assignment == null || !assignment.isValid()) return false;
+    GoReferenceExpression field = getFieldReference(element, assignment);
+    if (field == null) return false;
+    GoCompositeLit structDefinition = getStructDefinitionByField(field, assignment);
+    if (structDefinition == null) return false;
+    return isFieldUninitialized(field, structDefinition);
+  }
+
+  @Override
+  public void invoke(@NotNull Project project, Editor editor, @NotNull PsiElement element) throws IncorrectOperationException {
+    if (!element.isValid() || !element.isWritable()) return;
+    GoAssignmentStatement assignment = PsiTreeUtil.getParentOfType(element, GoAssignmentStatement.class, false);
+    if (assignment == null || !assignment.isValid()) return;
+    GoReferenceExpression field = getFieldReference(element, assignment);
+    if (field == null) return;
+    GoCompositeLit structDefinition = getStructDefinitionByField(field, assignment);
+    if (structDefinition == null || !isFieldUninitialized(field, structDefinition)) return;
+    GoExpression fieldValue = GoPsiImplUtil.getExpressionValue(assignment, field);
+    GoLiteralValue structFieldDefinitions = structDefinition.getLiteralValue();
+    if (fieldValue == null || structFieldDefinitions == null) return;
+    WriteCommandAction.runWriteCommandAction(project, () -> {
```

Comment:
@grenki I thought `WriteCommandAction` must use in all cases where happening Psi tree changes(what stuff like `PsiElement#add/delete` does), am i wrong?